### PR TITLE
bugfix(Whitebox): add small margin to prevent z-fighting when working with polygon manipulator

### PR DESCRIPTION
**Pull request by pollend**
this is not a perfect solution but it addresses the problem where the manipulator is z-fighting in whitebox. might make sense to make this a post processing affect i.e render the polygon separately and use the depth test to highlight the selected polygon maybe? 

if you get close to the mesh you can see where the selected region is floating by a small margin. this is better then the current situation where this will z-fight. 

image

### CODE REVIEW

**Comment by pollend** [comment]
ref: https://github.com/o3de/o3de/issues/8268

**Comment by AMZN-daimini** [comment]
Code looks good to me, but I will defer to people with more experience with the viewport before approving.

File: Gems/WhiteBox/Code/Source/Viewport/WhiteBoxPolygonTranslationModifier.cpp
Comment (hultonha) [inline Gems/WhiteBox/Code/Source/Viewport/WhiteBoxPolygonTranslationModifier.cpp:283]: nit: This should be `const` and we might want to move this closer to the first use (nearer line 294)

File: Gems/WhiteBox/Code/Source/Viewport/WhiteBoxPolygonTranslationModifier.h
Comment (hultonha) [inline Gems/WhiteBox/Code/Source/Viewport/WhiteBoxPolygonTranslationModifier.h:28]: update: Can this please be moved to an `AZ_CVAR` in the .cpp so we can tweak this potentially (there are other examples in the White Box Gem and the repo in general, I think this will be super useful to tweak if we need to)

Comment (hultonha) [reply Gems/WhiteBox/Code/Source/Viewport/WhiteBoxPolygonTranslationModifier.h:28]: I'd also recommend updating the name to `PolygonViewOverlapOffset` (as this isn't really related to the z/depth value)

File: Gems/WhiteBox/Code/Source/Viewport/WhiteBoxPolygonTranslationModifier.cpp
Comment (pollend) [inline Gems/WhiteBox/Code/Source/Viewport/WhiteBoxPolygonTranslationModifier.cpp:25]: what is the naming convention I don't think this is right?

Comment (hultonha) [reply Gems/WhiteBox/Code/Source/Viewport/WhiteBoxPolygonTranslationModifier.cpp:25]: Ah sorry yes we should now use `ed_` for editor specific settings (in a separate PR feel free to update all the White Box CVars to use `ed_` instead of `cl_`, we hadn't settled on that convention when these were first added unfortunately

File: Gems/WhiteBox/Code/Source/Viewport/WhiteBoxPolygonTranslationModifier.cpp
Comment (hultonha) [inline Gems/WhiteBox/Code/Source/Viewport/WhiteBoxPolygonTranslationModifier.cpp:300]: nit: Can we remove the trailing white space please, also I'd move line `286` down to near `300` to keep the variable lifetime as short as it needs to be

**Review by hultonha — APPROVED** [review]
I have a couple of minor comments but overall the change looks good. Would you be able to please include a video of what the effect looks like now, and try some more complex face configurations (be perturbing vertices that make up a polygon). I just want to make sure this looks good in different situations and not just the simple quad case. Thanks!

**Review by AMZN-daimini — APPROVED** [review]
Pending minor fixes noted by hultonha :)

**Review by hultonha — APPROVED** [review]
LGTM (if you can could you remove the trailing white space on line 288 😅 but it's not the end of the world, I'll kick off a build now 👍)

**Comment by cgalvan** [comment]
There are some unit tests that are failing from this change:

```
[2022-05-05T15:27:24.709Z] SettingsRegistryMergeUtils: Project path isn't set in the Settings Registry at "/Amazon/AzCore/Bootstrap/project_path". Project-related filepaths will be set relative to the executable directory

[2022-05-05T15:27:24.709Z] SettingsRegistryMergeUtils: Project path isn't set in the Settings Registry at "/Amazon/AzCore/Bootstrap/project_path". Project-related filepaths will be set relative to the executable directory

[2022-05-05T15:27:24.709Z] SettingsRegistryMergeUtils: Project path isn't set in the Settings Registry at "/Amazon/AzCore/Bootstrap/project_path". Project-related filepaths will be set relative to the executable directory

[2022-05-05T15:27:24.709Z] SettingsRegistryMergeUtils: 

[2022-05-05T15:27:24.709Z] ==================================================================

[2022-05-05T15:27:24.709Z] SettingsRegistryMergeUtils: Trace::Warning

[2022-05-05T15:27:24.709Z]  /data/workspace/o3de/Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.cpp(686): 'bool AZ::SettingsRegistryMergeUtils::MergeSettingsToRegistry_ConfigFile(AZ::SettingsRegistryInterface &, AZStd::string_view, const AZ::SettingsRegistryMergeUtils::ConfigParserSettings &)'

[2022-05-05T15:27:24.709Z] SettingsRegistryMergeUtils: Unable to open file "/data/workspace/o3de/build/linux/bin/profile/Cache/linux/user.cfg"

[2022-05-05T15:27:24.710Z] SettingsRegistryMergeUtils: ==================================================================

[2022-05-05T15:27:24.710Z] Logger: Network layer initialized

[2022-05-05T15:27:24.710Z] /data/workspace/o3de/Gems/WhiteBox/Code/Tests/WhiteBoxTestRailsAutomation.cpp:82: Failure

[2022-05-05T15:27:24.710Z] Value of: finalVertexLocalPositions

[2022-05-05T15:27:24.710Z] Expected: contains 8 values, where each value and its corresponding value in { (X: -0.5, Y: -0.5, Z: 0.5), (X: 20.5, Y: -0.5, Z: 0.5), (X: 20.5, Y: 0.5, Z: 0.5), (X: -0.5, Y: 0.5, Z: 0.5), (X: -0.5, Y: -0.5, Z: -0.5), (X: 20.5, Y: -0.5, Z: -0.5), (X: 20.5, Y: 0.5, Z: -0.5), (X: -0.5, Y: 0.5, Z: -0.5) } container is close tolerance 0.01

[2022-05-05T15:27:24.710Z]   Actual: { (X: -0.5, Y: -0.5, Z: 0.5), (X: 20.4854, Y: -0.5, Z: 0.5), (X: 20.4854, Y: 0.5, Z: 0.5), (X: -0.5, Y: 0.5, Z: 0.5), (X: -0.5, Y: -0.5, Z: -0.5), (X: 20.4854, Y: -0.5, Z: -0.5), (X: 20.4854, Y: 0.5, Z: -0.5), (X: -0.5, Y: 0.5, Z: -0.5) }, where the value pair ((X: 20.4854, Y: -0.5, Z: 0.5), (X: 20.5, Y: -0.5, Z: 0.5)) at index #1 don't match

[2022-05-05T15:27:24.710Z] Logger: Network layer shut down

[2022-05-05T15:27:24.710Z] [  FAILED  ] TestRailsAutomationFixture.C28798184_PressAndDragOnAPolygonTranslatesItAlongTheSurfaceNormal (970 ms)

[2022-05-05T15:27:24.710Z] [ RUN      ] TestRailsAutomationFixture.C28798192_PressAndDragOutwardsFromAPolygonWithCtrlHeldWillExtrudeThePolygon

[2022-05-05T15:27:24.710Z] #bytes written: 1478

[2022-05-05T15:27:24.710Z] SettingsRegistryMergeUtils: Project path isn't set in the Settings Registry at "/Amazon/AzCore/Bootstrap/project_path". Project-related filepaths will be set relative to the executable directory

[2022-05-05T15:27:24.710Z] SettingsRegistryMergeUtils: Project path isn't set in the Settings Registry at "/Amazon/AzCore/Bootstrap/project_path". Project-related filepaths will be set relative to the executable directory

[2022-05-05T15:27:24.710Z] SettingsRegistryMergeUtils: Project path isn't set in the Settings Registry at "/Amazon/AzCore/Bootstrap/project_path". Project-related filepaths will be set relative to the executable directory

[2022-05-05T15:27:24.710Z] SettingsRegistryMergeUtils: 

[2022-05-05T15:27:24.710Z] ==================================================================

[2022-05-05T15:27:24.710Z] SettingsRegistryMergeUtils: Trace::Warning

[2022-05-05T15:27:24.710Z]  /data/workspace/o3de/Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.cpp(686): 'bool AZ::SettingsRegistryMergeUtils::MergeSettingsToRegistry_ConfigFile(AZ::SettingsRegistryInterface &, AZStd::string_view, const AZ::SettingsRegistryMergeUtils::ConfigParserSettings &)'

[2022-05-05T15:27:24.710Z] SettingsRegistryMergeUtils: Unable to open file "/data/workspace/o3de/build/linux/bin/profile/Cache/linux/user.cfg"

[2022-05-05T15:27:24.710Z] SettingsRegistryMergeUtils: ==================================================================

[2022-05-05T15:27:24.710Z] Logger: Network layer initialized

[2022-05-05T15:27:24.710Z] /data/workspace/o3de/Gems/WhiteBox/Code/Tests/WhiteBoxTestRailsAutomation.cpp:149: Failure

[2022-05-05T15:27:24.710Z] Value of: finalVertexLocalPositions

[2022-05-05T15:27:24.710Z] Expected: contains 12 values, where each value and its corresponding value in { (X: -0.5, Y: -0.5, Z: 0.5), (X: 0.5, Y: -0.5, Z: 0.5), (X: 0.5, Y: 0.5, Z: 0.5), (X: -0.5, Y: 0.5, Z: 0.5), (X: -0.5, Y: -0.5, Z: -0.5), (X: 0.5, Y: -0.5, Z: -0.5), (X: 0.5, Y: 0.5, Z: -0.5), (X: -0.5, Y: 0.5, Z: -0.5), (X: 20.5, Y: -0.5, Z: -0.5), (X: 20.5, Y: 0.5, Z: -0.5), (X: 20.5, Y: 0.5, Z: 0.5), (X: 20.5, Y: -0.5, Z: 0.5) } container is close tolerance 0.01

[2022-05-05T15:27:24.710Z]   Actual: { (X: -0.5, Y: -0.5, Z: 0.5), (X: 0.5, Y: -0.5, Z: 0.5), (X: 0.5, Y: 0.5, Z: 0.5), (X: -0.5, Y: 0.5, Z: 0.5), (X: -0.5, Y: -0.5, Z: -0.5), (X: 0.5, Y: -0.5, Z: -0.5), (X: 0.5, Y: 0.5, Z: -0.5), (X: -0.5, Y: 0.5, Z: -0.5), (X: 20.4854, Y: -0.5, Z: -0.5), (X: 20.4854, Y: 0.5, Z: -0.5), (X: 20.4854, Y: 0.5, Z: 0.5), (X: 20.4854, Y: -0.5, Z: 0.5) }, where the value pair ((X: 20.4854, Y: -0.5, Z: -0.5), (X: 20.5, Y: -0.5, Z: -0.5)) at index #8 don't match

[2022-05-05T15:27:24.710Z] Logger: Network layer shut down

[2022-05-05T15:27:24.710Z] [  FAILED  ] TestRailsAutomationFixture.C28798192_PressAndDragOutwardsFromAPolygonWithCtrlHeldWillExtrudeThePolygon (1083 ms)

[2022-05-05T15:27:24.710Z] [ RUN      ] TestRailsAutomationFixture.C28798192_PressAndDragOnAPolygonScaleModifierScalesTheVerticesAboutThePolygonCenter

[2022-05-05T15:27:24.710Z] #bytes written: 929

[2022-05-05T15:27:24.710Z] SettingsRegistryMergeUtils: Project path isn't set in the Settings Registry at "/Amazon/AzCore/Bootstrap/project_path". Project-related filepaths will be set relative to the executable directory

[2022-05-05T15:27:24.710Z] SettingsRegistryMergeUtils: Project path isn't set in the Settings Registry at "/Amazon/AzCore/Bootstrap/project_path". Project-related filepaths will be set relative to the executable directory

[2022-05-05T15:27:24.710Z] SettingsRegistryMergeUtils: Project path isn't set in the Settings Registry at "/Amazon/AzCore/Bootstrap/project_path". Project-related filepaths will be set relative to the executable directory

[2022-05-05T15:27:24.710Z] SettingsRegistryMergeUtils: 

[2022-05-05T15:27:24.710Z] ==================================================================

[2022-05-05T15:27:24.710Z] SettingsRegistryMergeUtils: Trace::Warning

[2022-05-05T15:27:24.710Z]  /data/workspace/o3de/Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.cpp(686): 'bool AZ::SettingsRegistryMergeUtils::MergeSettingsToRegistry_ConfigFile(AZ::SettingsRegistryInterface &, AZStd::string_view, const AZ::SettingsRegistryMergeUtils::ConfigParserSettings &)'

[2022-05-05T15:27:24.710Z] SettingsRegistryMergeUtils: Unable to open file "/data/workspace/o3de/build/linux/bin/profile/Cache/linux/user.cfg"

[2022-05-05T15:27:24.710Z] SettingsRegistryMergeUtils: ==================================================================

[2022-05-05T15:27:24.710Z] Logger: Network layer initialized

[2022-05-05T15:27:24.710Z] Logger: Network layer shut down

[2022-05-05T15:27:24.710Z] [       OK ] TestRailsAutomationFixture.C28798192_PressAndDragOnAPolygonScaleModifierScalesTheVerticesAboutThePolygonCenter (943 ms)

[2022-05-05T15:27:24.710Z] [ RUN      ] TestRailsAutomationFixture.C28798193_PressAndDragInwardsFromAPolygonWithCtrlHeldWillImpressThePolygon

[2022-05-05T15:27:24.710Z] #bytes written: 929

[2022-05-05T15:27:24.710Z] #bytes written: 1478

[2022-05-05T15:27:24.710Z] SettingsRegistryMergeUtils: Project path isn't set in the Settings Registry at "/Amazon/AzCore/Bootstrap/project_path". Project-related filepaths will be set relative to the executable directory

[2022-05-05T15:27:24.710Z] SettingsRegistryMergeUtils: Project path isn't set in the Settings Registry at "/Amazon/AzCore/Bootstrap/project_path". Project-related filepaths will be set relative to the executable directory

[2022-05-05T15:27:24.710Z] SettingsRegistryMergeUtils: Project path isn't set in the Settings Registry at "/Amazon/AzCore/Bootstrap/project_path". Project-related filepaths will be set relative to the executable directory

[2022-05-05T15:27:24.710Z] SettingsRegistryMergeUtils: 

[2022-05-05T15:27:24.710Z] ==================================================================

[2022-05-05T15:27:24.710Z] SettingsRegistryMergeUtils: Trace::Warning

[2022-05-05T15:27:24.710Z]  /data/workspace/o3de/Code/Framework/AzCore/AzCore/Settings/SettingsRegistryMergeUtils.cpp(686): 'bool AZ::SettingsRegistryMergeUtils::MergeSettingsToRegistry_ConfigFile(AZ::SettingsRegistryInterface &, AZStd::string_view, const AZ::SettingsRegistryMergeUtils::ConfigParserSettings &)'

[2022-05-05T15:27:24.710Z] SettingsRegistryMergeUtils: Unable to open file "/data/workspace/o3de/build/linux/bin/profile/Cache/linux/user.cfg"

[2022-05-05T15:27:24.710Z] SettingsRegistryMergeUtils: ==================================================================

[2022-05-05T15:27:24.710Z] Logger: Network layer initialized

[2022-05-05T15:27:24.710Z] /data/workspace/o3de/Gems/WhiteBox/Code/Tests/WhiteBoxTestRailsAutomation.cpp:336: Failure

[2022-05-05T15:27:24.710Z] Value of: finalVertexLocalPositions

[2022-05-05T15:27:24.710Z] Expected: contains 12 values, where each value and its corresponding value in { (X: -0.5, Y: -0.5, Z: 0.5), (X: 0.5, Y: -0.25, Z: 0.25), (X: 0.5, Y: 0.25, Z: 0.25), (X: -0.5, Y: 0.5, Z: 0.5), (X: -0.5, Y: -0.5, Z: -0.5), (X: 0.5, Y: -0.25, Z: -0.25), (X: 0.5, Y: 0.25, Z: -0.25), (X: -0.5, Y: 0.5, Z: -0.5), (X: 20.5, Y: -0.25, Z: -0.25), (X: 20.5, Y: 0.25, Z: -0.25), (X: 20.5, Y: 0.25, Z: 0.25), (X: 20.5, Y: -0.25, Z: 0.25) } container is close tolerance 0.01

[2022-05-05T15:27:24.710Z]   Actual: { (X: -0.5, Y: -0.5, Z: 0.5), (X: 0.5, Y: -0.248746, Z: 0.248746), (X: 0.5, Y: 0.248746, Z: 0.248746), (X: -0.5, Y: 0.5, Z: 0.5), (X: -0.5, Y: -0.5, Z: -0.5), (X: 0.5, Y: -0.248746, Z: -0.248746), (X: 0.5, Y: 0.248746, Z: -0.248746), (X: -0.5, Y: 0.5, Z: -0.5), (X: 20.4854, Y: -0.248746, Z: -0.248746), (X: 20.4854, Y: 0.248746, Z: -0.248746), (X: 20.4854, Y: 0.248746, Z: 0.248746), (X: 20.4854, Y: -0.248746, Z: 0.248746) }, where the value pair ((X: 20.4854, Y: -0.248746, Z: -0.248746), (X: 20.5, Y: -0.25, Z: -0.25)) at index #8 don't match

[2022-05-05T15:27:24.710Z] Logger: Network layer shut down

[2022-05-05T15:27:24.710Z] [  FAILED  ] TestRailsAutomationFixture.C28798193_PressAndDragInwardsFromAPolygonWithCtrlHeldWillImpressThePolygon (1080 ms)

[2022-05-05T15:27:24.710Z] [----------] 4 tests from TestRailsAutomationFixture (4076 ms total)
```

**Comment by pollend** [comment]
the margin seems really small might be because the z offset I add is like .002 so its complaining that -0.25 != -0.248746 do you think the test should be adjusted for this @hultonha

**Comment by pollend** [comment]
@hultonha I knew changing the geometry changed the behavior of the manipulator so instead I moved that logic to draw pass so it shouldn't affect the tests.

File: Gems/WhiteBox/Code/Source/Viewport/WhiteBoxManipulatorViews.h
Comment (hultonha) [inline Gems/WhiteBox/Code/Source/Viewport/WhiteBoxManipulatorViews.h:39]: update: Can we give this a more descriptive name instead of just transform, as this could be confusing returning to this as it's really just an offset, could we call this `m_faceOffsetTransform` perhaps? (and add a comment to explain what it's used for). Thanks!

Comment (hultonha) [reply Gems/WhiteBox/Code/Source/Viewport/WhiteBoxManipulatorViews.h:39]: or even better similar to the CVar name - `m_polygonViewOverlapOffset` (and just omit `transform` as it's the type, we don't always need to include the name of the type as we're statically typed anyway 🙂)

File: Gems/WhiteBox/Code/Source/Viewport/WhiteBoxPolygonTranslationModifier.cpp
Comment (hultonha) [inline Gems/WhiteBox/Code/Source/Viewport/WhiteBoxPolygonTranslationModifier.cpp:309]: uber nit: can we please leave the line break after the closing brace please, just helps with separating blocks out

**Review by hultonha — APPROVED** [review]
Good fix! (and this is why we have the tests too! 😄) Looks like a good plan to me, I just have a couple of small suggestions, once they're updated I can kick off another build 👍

**Comment by hultonha** [comment]
@pollend we're still waiting on this PR to be merged https://github.com/o3de/o3de/pull/9423, there's a DCO issue we've run into that we're waiting to be resolved, will let you know when it's merged :+1:

**Comment by hultonha** [comment]
Just kicked off another build @pollend 👍

**Comment by hultonha** [comment]
> Got a green AR! So I've merged this fix, thanks @pollend !

Awesome thanks @cgalvan! 😄

**Pull request merged**
